### ISSUE TITLE
CKEditor textarea update on Ajax form submit

### ISSUE THREAD

**Issue by bigin** [GetSimpleCMS/GetSimpleCMS#1291]
Edit page menu on current dev:

```javascript
save_htmleditors = function() {
    $('.html_edit').each(function(i, textarea) {
...
```
It's not a flexible solution. What if the plugins want to implement custom editors within the edit-page menu?  

Better you go through all instances of CKEditor:

```javascript
for(var i in CKEDITOR.instances) {
    CKEDITOR.instances[i].updateElement(); 
}
```

**Comment by tablatronix** [comment]
What kind of custom editors?
Like a ckeditor with custom config or something else?

**Comment by tablatronix** [comment]
I added an exclusion for .custom class, editors will save but wont have htmledit cke added automatically. Not tested

**Comment by bigin** [comment]
yes I meant CKEditor.

**Comment by bigin** [comment]
And why not make it dynamic?

**Comment by bigin** [comment]
Do it your way, I can manage without it, it was just a suggestion.

**Comment by tablatronix** [comment]
dynamic how?
I am asking for the exact use case so I can design around it, I have to add support for non ckeditor editors in next version also.

I can add event emit for save all, etc

**Comment by tablatronix** [comment]
- [ ] add code_edit .custom exclusion 
 tested

**Comment by bigin** [comment]
yes event sounds good

**Comment by tablatronix** [comment]
- [ ] add event triggers to save_all_editors, possible async save ready problems?

**Comment by bigin** [comment]
shouldn't be a problem, I don't know exactly.

**Comment by tablatronix** [comment]
Its a start for now, should be enough to hack in custom ckeditors.

Adding support to custom cke for the other stuff like uploads, page select is not absctracted enough to use, It is almost easier to just use the built in editors via html_edit and then modify them on instance ready listeners, so you can keep those features.

**Comment by bigin** [comment]
Yeah should do it. I'll check it out when I get a chance, thanks.
